### PR TITLE
Fix issue in table iteration

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/39169

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TableUtils.java [hunk 1]
```diff
@@ -96,6 +96,8 @@ public static Long hash(Object obj, Node parent) {
             } else {
                 return (long) obj.hashCode();
             }
+        } else if (obj instanceof Long) {
```
Comment (gimantha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TableUtils.java:99]: I suspect that the same thing can happen for the floats also. Shall we add test cases for all basic data types? internally it seems the same algo is used for java double values to calculate the hashcode

**Review by hasithaa — APPROVED** [review]

**Comment by hasithaa** [comment]
Do we have test cases covering each type hashes?

**Comment by hasithaa** [comment]
@KavinduZoysa Please send a PR to Update 3.2 branch.

**Comment by KavinduZoysa** [comment]
> #39169

This is the new PR https://github.com/ballerina-platform/ballerina-lang/pull/39184

**Pull request merged**